botocore/parsers.py

```diff
@@ -707,8 +720,10 @@ def _do_error_parse(self, response, shape):
             # code has a couple forms as well:
             # * "com.aws.dynamodb.vAPI#ProvisionedThroughputExceededException"
             # * "ResourceNotFoundException"
+            if ':' in code:
+                code = code.split(':', 1)[0]
             if '#' in code:
-                code = code.rsplit('#', 1)[1]
+                code = code.split('#', 1)[1]
```

Comment:
This is semantically different. Are we enforcing that the code is everything following the first `#`?